interp/notation.ml

```diff
@@ -633,17 +633,27 @@ exception NotAValidPrimToken
     what it means for a term to be ground / to be able to be
     considered for parsing. *)
 
-let constr_of_globref allow_constant env sigma = function
+let is_primitive_type c =
+  let is_this_ref name c =
+    if Coqlib.has_ref name
+    then GlobRef.equal c (Coqlib.lib_ref name)
+    else false
+  in
+  is_this_ref "num.int63.type" c || is_this_ref "num.float.type" c || is_this_ref "core.parray.type" c
```

Comment:
Why not look in the env instead?

Reply:
Then you don't need to add a registration.

Reply:
Because I did cargo-cult this a bit to much. this should probably also fix the performance issue with the complexity/pretyping test.

Reply:
What perf issue is that?

Reply:
on my machine, that test now takes 162 instead of the allow 150 units, which i think is due to tha fact that now, whenever something is printed, every constant needs some lookups in the environment (or before, every constant performed several calls to Coqlib.find_ref.)

I also noticed that `Environ.is_primitive` apperently does not do what i expected there, which I'll have to inverstigate further.

Reply:
Do I have to use the retroknowledge as in Environ.is_array_type to test is the constant is primitive?
It seems that the primitive type constants are opaque and is_primitive does maybe never work?

Reply:
looks like it